services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/Consumer.java

```diff
@@ -81,4 +115,52 @@ protected void handle(ConsumerRecord<String, String> record) {
         logger.trace("received message: {} - {}", record.offset(), record.value());
         handlersPool.execute(handlerFactory.produce(record));
     }
+
+    @VisibleForTesting
+    static class KafkaOffsetRegistry {
+        private final KafkaConsumer<String, String> consumer;
+        private final long autoCommitInterval;
+        private final Map<TopicPartition, Long> partitionToUncommittedOffset = new HashMap<>();
+        private long lastCommitTime;
+
+        KafkaOffsetRegistry(KafkaConsumer<String, String> consumer, long autoCommitInterval) {
+            this.consumer = requireNonNull(consumer);
+            checkArgument(autoCommitInterval > 0, "autoCommitInterval must be positive");
+            this.autoCommitInterval = autoCommitInterval;
+
+            lastCommitTime = System.currentTimeMillis();
+        }
+
+        /**
+         * Add the record's offset to the registry and perform a commit
+         * if more than autoCommitInterval ms passed since the last commit.
+         */
+        void addAndCommit(ConsumerRecord<String, String> record) {
+            TopicPartition partition = new TopicPartition(record.topic(), record.partition());
+            partitionToUncommittedOffset.put(partition, record.offset());
+
+            // commit offsets of processed messages
+            if ((System.currentTimeMillis() - lastCommitTime) >= autoCommitInterval) {
+                commitOffsets();
+            }
+        }
+
+        /**
+         * Commits the offsets added since the last commit.
+         */
+        void commitOffsets() {
+            if (!partitionToUncommittedOffset.isEmpty()) {
```

Comment:
excess check

Reply:
I don't think so, `consumer.commitSync` even for empty map performs some actions: acquiring/releasing a lock, checking the internal state. Even more -  why should we bother creating an empty map object when we can skip all that?